File: app/src/main/java/com/test/xyz/demo/presentation/common/BaseFragment.java
package com.test.xyz.demo.presentation.common;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

import com.test.xyz.demo.domain.model.github.GitHubRepo;

import java.util.ArrayList;
import java.util.List;

public abstract class BaseFragment extends Fragment {
    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        setRetainInstance(true);
    }

    protected List<String> mapRepoList(List<GitHubRepo> gitHubRepos) {
        List<String> values = new ArrayList<>();
        for (int i = 0; i < gitHubRepos.size(); ++i) {
            values.add(gitHubRepos.get(i).name);
        }
        return values;
    }
}

File: app/src/main/kotlin/com/test/xyz/demo/presentation/common/BaseFragment.kt
package com.test.xyz.demo.presentation.common

import android.os.Bundle
import android.support.v4.app.Fragment
import com.test.xyz.demo.domain.model.github.GitHubRepo
import java.util.*

abstract class BaseFragment : Fragment() {
    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        retainInstance = true
    }

    protected fun mapRepoList(gitHubRepos: List<GitHubRepo>): List<String> {
        val values = ArrayList<String>()
        for (i in gitHubRepos.indices) {
            values.add(gitHubRepos[i].name)
        }
        return values
    }
}
